store/bsdb/bucket.go

```diff
@@ -77,6 +77,22 @@ func (b *BsDBImpl) GetUserBucketsCount(accountID common.Address) (int64, error)
 		err   error
 	)
 
-	err = b.db.Table((&Bucket{}).TableName()).Select("count(1)").Take(&count, "owner_address = ?", accountID).Error
+	err = b.db.Table((&Bucket{}).TableName()).Select("count(1)").Take(&count, "owner = ?", accountID).Error
 	return count, err
 }
+
+// ListExpiredBucketsBySp lists expired buckets
+func (b *BsDBImpl) ListExpiredBucketsBySp(createAt int64, primarySpAddress string) ([]*Bucket, error) {
+	var (
+		buckets []*Bucket
+		err     error
+	)
+
+	err = b.db.Table((&Bucket{}).TableName()).
+		Select("*").
+		Where("primary_sp_address = ? and status = 'BUCKET_STATUS_CREATED' and create_time < ? and removed = false", common.HexToAddress(primarySpAddress), createAt).
```

Comment:
1.  Is it be possible to make the status as a parameter?  so that we can support more cases.
2. I remember you mentioned you will return the first 1000 rows. Do we still this behavior ?  How do we handle pagination ?

Reply:
For 1, since bucket status only have 2 kinds: BUCKET_STATUS_CREATED, BUCKET_STATUS_DISCONTINUED, we only need BUCKET_STATUS_CREATED for the purpose of this specific query. If we will ever need to filter BUCKET_STATUS_DISCONTINUED status, "removed" in this query shall also become a parameter, and I feel the difference is large enough to be separate out as a stand alone new method.

For 2. returning first 1000 rows is the behavior of ListDeletedObjectsByBlockNumberRange. For this method, we will develop the pagination strategy along with GetUserBuckets in next phase.

Reply:
Update: Pass in a new parameter for limit, can be within the range of 1-1000